one-ui/src/main/java/com/marklogic/hub/oneui/exceptions/CustomExceptionHandler.java

```diff
@@ -97,10 +97,10 @@ private String httpStatusSuggestion(HttpStatus httpStatus) {
 
     private String exceptionSuggestion(Exception exception) {
         Throwable rootException = exception;
-        if (exception instanceof RuntimeException && exception.getCause() != null) {
+        if (exception.getCause() != null) {
             rootException = exception.getCause();
         }
-        if (rootException instanceof IOException) {
+        if (rootException instanceof IOException || exception instanceof IOException) {
```

Comment:
Oh okay - IOException is being used as way of indicating that the project directory is unavailable. I think IOException is too broad for that - I can definitely see some other feature in the future throwing an IOException that has nothing to do with the project directory.

I recommend creating an e.g. "ProjectDirectoryException" and catching any possible IOException from the code that tries to access the project directory. This code can then catch that exception without any concern that something else may throw the exception.